dev/io.openliberty.org.jboss.resteasy.common.ee10/src/org/jboss/resteasy/core/registry/SegmentNode.java

```diff
@@ -0,0 +1,603 @@
+package org.jboss.resteasy.core.registry;
```

Comment:
In cases where we really can't avoid overlays, it is a good idea to
a) make sure that every overlaid file has an appropriate copyright/license header, even if the original did not. In this case it would have been easy to copy from one of the other classes in the same package
b) add a comment which links back to the exact version of the original source file